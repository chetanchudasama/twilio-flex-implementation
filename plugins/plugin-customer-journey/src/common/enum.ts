export enum VulnerableCustomerStatus {
  suspected = 1,
  confirmed = 2,
}

export enum HistoryPanelMemberType {
  Customer = 1,
  Agent = 2,
}

export enum MessageDirectionType {
  Outbound = 1,
  Inbound = 2,
}

export enum ChannelType {
  SMS = 1,
  Whatsapp = 2,
}

export enum VehicleSortingOptions {
  PriceDescending = 2,
  PriceAscending = 3,
  AgeAscending = 6,
}

export enum QuoteSortingOptions {
  Price = 1,
  Distance = 2,
  Age = 3,
  AddedDate = 4,
  Mileage = 5,
}

export enum LoanCalculatorType {
  MonthlyRepayment = 1,
  LoanAmount = 2,
}

export enum DealerStateType {
  Approved = 1,
  Declined = 2,
  Unknown = 3,
  ApprovedForDCI = 4,
}

export enum PreferredNameType {
  FullName,
  FirstName,
  Other,
}
