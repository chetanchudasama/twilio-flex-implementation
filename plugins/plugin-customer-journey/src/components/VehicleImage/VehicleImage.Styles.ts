import {
  white,
  borderStyle,
  defaultSnackbarColor,
  darkGrey,
} from "@common/components";
import styled from "react-emotion";

export const VehicleImageStyles = styled("div")({
  width: "100%",
  height: "100%",
  border: borderStyle,
  position: "relative",
  backgroundColor: defaultSnackbarColor,
  borderRadius: "10px 0px 0px 3px",

  ".vehicle-img": {
    width: "100%",
    height: "100%",
    maxWidth: "100%",
    maxHeight: "100%",
    objectFit: "contain",
    cursor: "pointer",
  },
  ".image-count": {
    position: "absolute",
    left: "0",
    top: "0",
    display: "flex",
    justifyContent: "center",
    padding: "10px",
    color: white,
    background: darkGrey,
    cursor: "pointer",
    borderRadius: "10px 0px 5px 0px",
  },
  ".icon": {
    fontSize: "18px",
  },
  ".image-count .icon": {
    paddingRight: "8px",
  },
  ".image-missing": {
    width: "100%",
    height: "100%",
  },
});
