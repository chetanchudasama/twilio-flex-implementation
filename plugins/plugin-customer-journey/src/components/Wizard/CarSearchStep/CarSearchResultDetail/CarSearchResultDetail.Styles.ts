import {
  black,
  darkGrey,
  mediumGrey,
  secondary,
  white,
  darkPurpleHover,
  borderStyle,
} from "@common/components";
import styled from "react-emotion";

export const CarSearchResultDetailStyles = styled("div")({
  position: "relative",
  border: borderStyle,
  margin: "10px",
  height: "auto",
  maxHeight: "375px",
  borderRadius: 10,

  ".car-search-detail": {
    padding: "14px",
    borderBottom: borderStyle,
    height: "275px",
  },
  ".dealer-detail": {
    padding: "6px 14px 14px 14px",
    height: "100px",
    alignItems: "center",
  },
  ".car-price": {
    paddingTop: "10px",
    color: secondary,
    fontWeight: 600,
    fontSize: "18px",
  },
  ".car-model": {
    fontSize: "16px",
    fontWeight: 600,
  },
  ".view-more-detail": {
    marginBottom: "5px",
  },
  ".view-more-detail-link": {
    color: black,
    fontSize: "16px",
    textDecoration: "underline",
    padding: 0,
    textTransform: "capitalize",
  },
  ".view-more-detail-link:hover": {
    background: "none",
    textDecoration: "underline",
  },
  ".offer-best-match": {
    border: borderStyle,
    margin: "0 0 8px 5px",
    color: darkGrey,
    cursor: "pointer",
    borderRadius: 3,
  },
  ".offer-unavailable": {
    border: borderStyle,
    margin: "0 0 8px 5px",
    color: mediumGrey,
    borderRadius: 3,
  },
  ".offer-detail": {
    border: borderStyle,
    margin: "0 0 8px 5px",
    color: darkGrey,
    cursor: "pointer",
    borderRadius: 3,
  },
  ".offer-selected": {
    border: `2px solid${secondary}`,
    margin: "0 0 8px 5px",
    color: darkGrey,
    cursor: "pointer",
    borderRadius: 3,
  },
  ".offer-best-match-heading": {
    textAlign: "center",
    padding: "5px",
    borderBottom: borderStyle,
    cursor: "pointer",
  },
  ".offer-best-match-selected": {
    background: secondary,
    textAlign: "center",
    color: white,
    padding: "5px",
    cursor: "pointer",
  },
  ".offer-detail-content": {
    padding: "6px",
    textAlign: "center",
    cursor: "pointer",
    borderRadius: 3,
  },
  ".offer-unavailable .offer-detail-content": {
    padding: "6px",
    paddingBottom: "10px",
    cursor: "default",
    borderRadius: 3,
  },
  ".offer-term": {
    fontSize: "14px",
    paddingBottom: "5px",
  },
  ".offer-amount": {
    fontSize: "16px",
    lineHeight: "20px",
    fontWeight: 600,
  },
  ".offer-pm": {
    fontSize: "14px",
    lineHeight: "22px",
    paddingLeft: "5px",
  },
  ".offer-price": {
    display: "flex",
    justifyContent: "center",
  },
  ".dealer-fee": {
    textAlign: "center",
    wordBreak: "break-all",
    margin: "5px 0 0 5px",
    color: darkGrey,
  },
  ".quote-btn": {
    backgroundColor: secondary,
    color: white,
    "&:hover": {
      backgroundColor: darkPurpleHover,
    },
    margin: "5px 5px 5px 0",
    height: "36px",
    padding: "5px 15px",
    borderRadius: 3,
  },
  ".secondary-button": {
    color: secondary,
    margin: "5px 5px 5px 0",
    height: "36px",
    padding: "5px 15px",
    borderRadius: 3,
  },
  ".car-specification": {
    listStyleType: "none",
    margin: "5px 0",
    padding: 0,
    width: "100%",
    maxHeight: "36px",
    overflow: "hidden",
  },
  ".car-specification li": {
    fontSize: "14px",
    display: "inline-block",
    position: "relative",
    marginBottom: "4px",
    lineHeight: "14px",
    "&:after": {
      content: "' |'",
      margin: "0 2px",
    },
  },
  ".car-specification li:last-child:after": { content: "''" },
  ".vehicle-img-container": {
    height: "373px",
    width: "100%",
  },
});
