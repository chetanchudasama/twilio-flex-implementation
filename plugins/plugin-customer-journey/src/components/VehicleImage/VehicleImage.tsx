import { ImageCarousel, CarNotFoundIcon } from "@common/components";
import React, { useState, useEffect } from "react";
import CameraAltIcon from "@material-ui/icons/CameraAlt";
import { VehicleImageStyles } from "./VehicleImage.Styles";

export interface VehicleImageProps {
  image: string;
  imageList?: string[];
  imagesCount?: number;
  showImageCount?: boolean;
}

export const VehicleImage: React.FC<VehicleImageProps> = ({
  image,
  imageList,
  imagesCount,
  showImageCount,
}) => {
  const [isImageUrlValid, setIsImageUrlValid] = useState(true);
  const [showCarousel, setShowCarousel] = useState<boolean>(false);

  const hasImages = imageList && imageList.length > 0;

  useEffect(() => {
    const imageExists = (
      url: string,
      callback: (isImageExists: boolean) => void
    ) => {
      const img: HTMLImageElement = new Image();
      img.onload = () => {
        callback(true);
      };
      img.onerror = () => {
        callback(false);
      };
      img.src = url;
    };

    if (image) {
      imageExists(image, (isImageExists) => {
        setIsImageUrlValid(isImageExists);
      });
    } else {
      setIsImageUrlValid(false);
    }
  }, [image]);

  return (
    <>
      <VehicleImageStyles
        onClick={() =>
          isImageUrlValid && hasImages ? setShowCarousel(true) : {}
        }
      >
        {image && isImageUrlValid ? (
          <>
            <img src={image} className="vehicle-img" alt="" />
            {isImageUrlValid && showImageCount && (
              <div className="image-count">
                <CameraAltIcon className="icon" />
                {imagesCount ?? 0}
              </div>
            )}
          </>
        ) : (
          <div className="image-missing">
            <CarNotFoundIcon />
          </div>
        )}
      </VehicleImageStyles>
      {showCarousel && hasImages && (
        <ImageCarousel
          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
          images={imageList!}
          showCarousel={showCarousel}
          handleClose={() => setShowCarousel(false)}
        />
      )}
    </>
  );
};
