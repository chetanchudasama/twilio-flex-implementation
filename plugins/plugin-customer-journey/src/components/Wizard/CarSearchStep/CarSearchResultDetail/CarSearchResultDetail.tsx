import { OfferInformationModel } from "models/OfferInformationModel";
import React, { useMemo, useState } from "react";

import { secondary, Shared, WhatsAppIcon } from "@common/components";
import { Button } from "@material-ui/core";
import Grid from "@material-ui/core/Grid";
import ChatBubbleOutlineIcon from "@material-ui/icons/ChatBubbleOutline";
import FavoriteIcon from "@material-ui/icons/Favorite";
import FavoriteBorderRoundedIcon from "@material-ui/icons/FavoriteBorderRounded";

import { ChannelType, DealerStateType } from "../../../../common/enum";
import { QuoteDetailModel } from "../../../../models/QuoteDetailModel";
import { VehicleDetailModel } from "../../../../models/VehicleDetailModel";
import { VehicleSummaryDetailModel } from "../../../../models/VehicleSummaryDetailModel";
import CarDealerInformation from "../../../CarDealerInformation/CarDealerInformation.Container";
import CarDetailViewDialog from "../../../CarDetailViewDialog/CarDetailViewDialog.Container";
import { VehicleImage } from "../../../VehicleImage/VehicleImage";
import QuoteDialog from "../../QuotesStep/QuoteDialog/QuoteDialog.Container";
import { CarSearchResultDetailStyles } from "./CarSearchResultDetail.Styles";

export interface CarSearchResultDetailProp {
  vehicleDetail: VehicleSummaryDetailModel | VehicleDetailModel;
  isCarSaved: boolean;
  sendCarDetail: (channelType: ChannelType) => void;
  saveCarDetail: () => void;
}

const CarSearchResultDetail: React.FC<CarSearchResultDetailProp> = ({
  vehicleDetail,
  isCarSaved,
  sendCarDetail,
  saveCarDetail,
}) => {
  const getCarMakeModel = () => {
    return Shared.getFormattedText(
      [vehicleDetail.make, vehicleDetail.model],
      " | "
    );
  };
  const [selectedOfferTerm, setSelectedOfferTerm] = useState(36);
  const [openCarDetailDialog, setOpenCarDetailDialog] =
    useState<boolean>(false);
  const [openCreateQuoteDialog, setOpenCreateQuoteDialog] = useState(false);
  const [quoteDetail, setQuoteDetail] = useState(new QuoteDetailModel());

  // TODO: remove once fetched from API
  const offerInformation: OfferInformationModel[] = [
    {
      term: 36,
      monthlyAmount: 2000,
      bestMatch: true,
      available: true,
    },
    {
      term: 48,
      monthlyAmount: 5000,
      bestMatch: false,
      available: false,
    },
    {
      term: 60,
      monthlyAmount: 8000,
      bestMatch: false,
      available: true,
    },
  ];

  const getOfferClassName = (offer: OfferInformationModel) => {
    if (offer.term === selectedOfferTerm) {
      return "offer-selected";
    }

    if (offer.bestMatch) {
      return "offer-best-match";
    }

    if (!offer.available) {
      return "offer-unavailable";
    }

    return "offer-detail";
  };

  const openQuoteDialog = () => {
    // assign vehicle detail to quote detail
    const newQuoteDetail = new QuoteDetailModel(vehicleDetail);
    newQuoteDetail.term =
      Shared.termList.find((t) => t.id === selectedOfferTerm)?.id ||
      Shared.termList[0].id;
    setQuoteDetail(newQuoteDetail);
    // open quote dialog
    setOpenCreateQuoteDialog(true);
  };

  const image = useMemo(() => {
    if (vehicleDetail instanceof VehicleSummaryDetailModel) {
      return vehicleDetail.image;
    }
    return vehicleDetail.images.length > 0 ? vehicleDetail.images[0] : "";
  }, [vehicleDetail]);

  const imagesCount = useMemo(() => {
    if (vehicleDetail instanceof VehicleSummaryDetailModel) {
      return vehicleDetail.imagesCount;
    }
    return vehicleDetail.images.length;
  }, [vehicleDetail]);

  const dealerAdminFee = useMemo(() => {
    if (vehicleDetail instanceof VehicleSummaryDetailModel) {
      return vehicleDetail.dealerAdminFee || 0;
    }
    return 0;
  }, [vehicleDetail]);

  const offerTermFunction = (offer: OfferInformationModel) => {
    if (offer.available) {
      setSelectedOfferTerm(offer.term);
    }
  };

  const disableSendMessageButton = useMemo(() => {
    return (
      !vehicleDetail.is247Cars ||
      vehicleDetail.dealer.state !== DealerStateType.Approved
    );
  }, [vehicleDetail.dealer.state, vehicleDetail.is247Cars]);

  return (
    <>
      <CarSearchResultDetailStyles>
        <Grid container>
          <Grid item xs={4} className="vehicle-img-container">
            <VehicleImage
              image={image}
              imageList={vehicleDetail.images}
              imagesCount={imagesCount}
              showImageCount
            />
          </Grid>
          <Grid item xs={8}>
            <Grid container className="car-search-detail">
              <Grid item xs={8}>
                <Grid container spacing={8} className="car-carousel">
                  <Grid item xs={12}>
                    <div className="car-price">
                      {Shared.getFormattedCurrencyValue(vehicleDetail.price)}
                    </div>
                  </Grid>
                  <Grid item xs={12}>
                    <div className="car-model">{getCarMakeModel()}</div>
                  </Grid>
                  <Grid item xs={12}>
                    <ul className="car-specification">
                      <li>
                        {`${Shared.getFormattedNumber(
                          vehicleDetail.mileage
                        )} miles`}
                      </li>
                      <li>{vehicleDetail.regNumber}</li>
                      <li>{vehicleDetail.regYear}</li>
                      <li>{vehicleDetail.trans}</li>
                      <li>{vehicleDetail.fuel}</li>
                    </ul>
                  </Grid>
                  <Grid item xs={12} className="view-more-detail">
                    <Button
                      className="view-more-detail-link"
                      onClick={() => setOpenCarDetailDialog(true)}
                      disableRipple
                    >
                      View More Detail
                    </Button>
                  </Grid>
                  <Grid item xs={12}>
                    <Button
                      variant="contained"
                      onClick={openQuoteDialog}
                      className="quote-btn"
                    >
                      Quote
                    </Button>
                    <Button
                      variant="outlined"
                      className="secondary-button"
                      onClick={saveCarDetail}
                    >
                      {isCarSaved ? (
                        <FavoriteIcon className="icon" />
                      ) : (
                        <FavoriteBorderRoundedIcon className="icon" />
                      )}
                    </Button>
                    <Button
                      variant="outlined"
                      className="secondary-button whatsapp-icon-button"
                      onClick={() => sendCarDetail(ChannelType.Whatsapp)}
                      disabled={disableSendMessageButton}
                    >
                      <WhatsAppIcon
                        color={!disableSendMessageButton ? secondary : ""}
                        opacity={disableSendMessageButton ? "0.26" : ""}
                      />
                    </Button>
                    <Button
                      variant="outlined"
                      className="secondary-button sms-icon-button"
                      onClick={() => sendCarDetail(ChannelType.SMS)}
                      disabled={disableSendMessageButton}
                    >
                      <ChatBubbleOutlineIcon className="icon" />
                    </Button>
                  </Grid>
                </Grid>
              </Grid>
              <Grid item xs={4}>
                {offerInformation.map((offer) => (
                  <React.Fragment key={offer.term}>
                    <div
                      className={getOfferClassName(offer)}
                      onClick={() => offerTermFunction(offer)}
                      onKeyDown={() => offerTermFunction(offer)}
                      role="button"
                      tabIndex={0}
                    >
                      {offer.bestMatch && (
                        <div
                          className={
                            offer.term === selectedOfferTerm
                              ? "offer-best-match-selected"
                              : "offer-best-match-heading"
                          }
                        >
                          Best Match
                        </div>
                      )}
                      <div className="offer-detail-content">
                        <div className="offer-term">
                          {offer.term} months from
                        </div>
                        <div className="offer-price">
                          {!offer.available ? (
                            <div className="offer-amount">Unavailable</div>
                          ) : (
                            <>
                              <div className="offer-amount">
                                {Shared.getFormattedCurrencyValue(
                                  offer.monthlyAmount
                                )}
                              </div>
                              <div className="offer-pm">/pm</div>
                            </>
                          )}
                        </div>
                      </div>
                    </div>
                  </React.Fragment>
                ))}
                <div className="dealer-fee">
                  {`Excl. ${Shared.getFormattedCurrencyValue(
                    dealerAdminFee
                  )} dealer fee`}
                </div>
              </Grid>
            </Grid>
            {/* <Grid container className="dealer-detail">
              <Grid item xs={12}>
                <CarDealerInformation
                  dealerInfo={vehicleDetail.dealer}
                  distance={vehicleDetail.distance!}
                />
              </Grid>
            </Grid> */}
          </Grid>
        </Grid>
      </CarSearchResultDetailStyles>
      {openCarDetailDialog && (
        <CarDetailViewDialog
          open={openCarDetailDialog}
          vehicleId={vehicleDetail.vehicleId}
          handleCloseDialog={() => setOpenCarDetailDialog(false)}
        />
      )}
      {openCreateQuoteDialog && (
        <QuoteDialog
          open={openCreateQuoteDialog}
          quoteDetail={quoteDetail}
          handleCloseDialog={() => setOpenCreateQuoteDialog(false)}
        />
      )}
    </>
  );
};

export default CarSearchResultDetail;
