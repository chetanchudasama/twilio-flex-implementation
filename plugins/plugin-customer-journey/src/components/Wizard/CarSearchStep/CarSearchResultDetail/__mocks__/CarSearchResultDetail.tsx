import React from "react";

import { CarSearchResultDetailProp } from "../CarSearchResultDetail";

const CarSearchResultDetail: React.FC<CarSearchResultDetailProp> = (props) => (
  <div />
);

export default CarSearchResultDetail;
