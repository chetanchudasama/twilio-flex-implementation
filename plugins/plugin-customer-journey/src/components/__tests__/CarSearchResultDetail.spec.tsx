import { Shared } from "@common/components";
import React from "react";
import { create } from "react-test-renderer";
import { mount } from "enzyme";
import FavoriteIcon from "@material-ui/icons/Favorite";
import FavoriteBorderRoundedIcon from "@material-ui/icons/FavoriteBorderRounded";
import { DealerStateType } from "../../common/enum";
import { DealerResponseModel } from "../../models/DealerResponseModel";
import { VehicleSummaryDetailModel } from "../../models/VehicleSummaryDetailModel";
import CarSearchResultDetail, {
  CarSearchResultDetailProp,
} from "../Wizard/CarSearchStep/CarSearchResultDetail/CarSearchResultDetail";
import { VehicleImage } from "../VehicleImage/VehicleImage";

jest.mock("../CarDetailViewDialog/CarDetailViewDialog.Container", () => {
  return {
    __esModule: true,
    default: () => {
      return <div className="car-detail-dialog">Car Detail View Dialog</div>;
    },
  };
});

jest.mock("../CarDealerInformation/CarDealerInformation.Container");
jest.mock("../CarDealerInformation/CarDealerInformation");

const props: CarSearchResultDetailProp = {
  vehicleDetail: Object.assign(new VehicleSummaryDetailModel(), {
    vehicleId: "48b14c4f-7928-49b1-aa15-bc8c1cad20ae",
    dealerAdminFee: 200,
    imagesCount: 1,
    image: "imageUrl",
    images: ["imageUrl"],
    make: "Bentley Motors",
    model: "Bentley Continental GT",
    price: 500000,
    mileage: 13,
    regNumber: "BM12853",
    regYear: 2016,
    trans: "Automatic",
    fuel: "Petrol",
    is247Cars: true,
    dealer: Object.assign(new DealerResponseModel(), {
      state: DealerStateType.Approved,
    }),
  }),
  sendCarDetail: jest.fn(() => {}),
  saveCarDetail: jest.fn(() => {}),
  isCarSaved: false,
};

describe("CarSearchResultDetail", () => {
  it("renders correctly", async () => {
    const wrapper = create(<CarSearchResultDetail {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it("should render props in UI", async () => {
    const wrapper = create(<CarSearchResultDetail {...props} />).root;

    expect(
      wrapper.findByProps({
        className: "dealer-fee",
      }).props.children
    ).toContain(
      (props.vehicleDetail as VehicleSummaryDetailModel).dealerAdminFee
    );
    expect(
      wrapper.findByProps({
        className: "car-specification",
      }).props.children.length
    ).toEqual(5);
    expect(
      wrapper.findByProps({
        className: "car-model",
      }).props.children
    ).toContain(
      Shared.getFormattedText(
        [props.vehicleDetail.make, props.vehicleDetail.model],
        " | "
      )
    );
  });

  it("should render VehicleImage component", async () => {
    const wrapper = mount(<CarSearchResultDetail {...props} />);

    expect(wrapper.find(VehicleImage).exists()).toBeTruthy();
  });

  it("when value of isCarSaved prop is false, FavoriteBorderRoundedIcon icon should be displayed", async () => {
    const wrapper = mount(<CarSearchResultDetail {...props} />);

    expect(wrapper.find(FavoriteBorderRoundedIcon).exists()).toBeTruthy();
  });

  it("when value of isCarSaved prop is true, FavoriteIcon icon should be displayed", async () => {
    props.isCarSaved = true;
    const wrapper = mount(<CarSearchResultDetail {...props} />);

    expect(wrapper.find(FavoriteIcon).exists()).toBeTruthy();
  });

  it("when view more detail link is clicked, CarDetailViewDialog component should be rendered", async () => {
    const wrapper = mount(<CarSearchResultDetail {...props} />);

    wrapper.find(".view-more-detail-link").at(1).simulate("click");

    expect(wrapper.find(".car-detail-dialog").exists()).toBeTruthy();
  });

  it("when dealer state is not Approved, SMS and Whatsapp buttons should be disabled", async () => {
    props.vehicleDetail.dealer.state = DealerStateType.Unknown;
    const wrapper = mount(<CarSearchResultDetail {...props} />);

    const whatsappButton = wrapper.find({
      className: "secondary-button whatsapp-icon-button",
    });

    const smsButton = wrapper.find({
      className: "secondary-button sms-icon-button",
    });

    expect(whatsappButton.last().getElement().props.disabled).toBeTruthy();
    expect(smsButton.last().getElement().props.disabled).toBeTruthy();
  });

  it("when car is not marked as is247Cars, SMS and Whatsapp buttons should be disabled", async () => {
    props.vehicleDetail.is247Cars = false;
    const wrapper = mount(<CarSearchResultDetail {...props} />);

    const whatsappButton = wrapper.find({
      className: "secondary-button whatsapp-icon-button",
    });

    const smsButton = wrapper.find({
      className: "secondary-button sms-icon-button",
    });

    expect(whatsappButton.last().getElement().props.disabled).toBeTruthy();
    expect(smsButton.last().getElement().props.disabled).toBeTruthy();
  });
});
