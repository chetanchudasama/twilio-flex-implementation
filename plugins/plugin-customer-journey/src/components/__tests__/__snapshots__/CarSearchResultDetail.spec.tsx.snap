// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarSearchResultDetail renders correctly 1`] = `
<div
  className="css-1jf3sw2"
>
  <div
    className="MuiGrid-container-1"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-4-33 vehicle-img-container"
    >
      <div
        className="css-1epo2pu"
        onClick={[Function]}
      >
        <img
          alt=""
          className="vehicle-img"
          src="imageUrl"
        />
        <div
          className="image-count"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-98 icon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="3.2"
            />
            <path
              d="M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
          1
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-8-37"
    >
      <div
        className="MuiGrid-container-1 car-search-detail"
      >
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-8-37"
        >
          <div
            className="MuiGrid-container-1 MuiGrid-spacing-xs-8-23 car-carousel"
          >
            <div
              className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
            >
              <div
                className="car-price"
              >
                Â£500,000
              </div>
            </div>
            <div
              className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
            >
              <div
                className="car-model"
              >
                Bentley Motors | Bentley Continental GT
              </div>
            </div>
            <div
              className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
            >
              <ul
                className="car-specification"
              >
                <li>
                  13 miles
                </li>
                <li>
                  BM12853
                </li>
                <li>
                  2016
                </li>
                <li>
                  Automatic
                </li>
                <li>
                  Petrol
                </li>
              </ul>
            </div>
            <div
              className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 view-more-detail"
            >
              <button
                className="MuiButtonBase-root-133 MuiButton-root-107 MuiButton-text-109 MuiButton-flat-112 view-more-detail-link"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiButton-label-108"
                >
                  View More Detail
                </span>
              </button>
            </div>
            <div
              className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
            >
              <button
                className="MuiButtonBase-root-133 MuiButton-root-107 MuiButton-contained-118 MuiButton-raised-121 quote-btn"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiButton-label-108"
                >
                  Quote
                </span>
                <span
                  className="MuiTouchRipple-root-136"
                />
              </button>
              <button
                className="MuiButtonBase-root-133 MuiButton-root-107 MuiButton-outlined-115 secondary-button"
                disabled={false}
                onBlur={[Function]}
                onClick={[MockFunction]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiButton-label-108"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-98 icon"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M0 0h24v24H0V0z"
                      fill="none"
                    />
                    <g>
                      <path
                        d="M19.66 3.99c-2.64-1.8-5.9-.96-7.66 1.1-1.76-2.06-5.02-2.91-7.66-1.1-1.4.96-2.28 2.58-2.34 4.29-.14 3.88 3.3 6.99 8.55 11.76l.1.09c.76.69 1.93.69 2.69-.01l.11-.1c5.25-4.76 8.68-7.87 8.55-11.75-.06-1.7-.94-3.32-2.34-4.28zM12.1 18.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
                      />
                    </g>
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-136"
                />
              </button>
              <button
                className="MuiButtonBase-root-133 MuiButton-root-107 MuiButton-outlined-115 secondary-button whatsapp-icon-button"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiButton-label-108"
                >
                  <svg
                    fontSize="22px"
                    height="1em"
                    viewBox="0 0 24 24"
                    width="1em"
                    x="0px"
                    y="0px"
                  >
                    <path
                      d="M16.75 13.96c.25.13.41.2.46.3.06.11.04.61-.21 1.18-.2.56-1.24 1.1-1.7 1.12-.46.02-.47.36-2.96-.73-2.49-1.09-3.99-3.75-4.11-3.92-.12-.17-.96-1.38-.92-2.61.05-1.22.69-1.8.95-2.04.24-.26.51-.29.68-.26h.47c.15 0 .36-.06.55.45l.69 1.87c.06.13.1.28.01.44l-.27.41-.39.42c-.12.12-.26.25-.12.5.12.26.62 1.09 1.32 1.78.91.88 1.71 1.17 1.95 1.3.24.14.39.12.54-.04l.81-.94c.19-.25.35-.19.58-.11l1.67.88M12 2a10 10 0 0 1 10 10 10 10 0 0 1-10 10c-1.97 0-3.8-.57-5.35-1.55L2 22l1.55-4.65A9.969 9.969 0 0 1 2 12 10 10 0 0 1 12 2m0 2a8 8 0 0 0-8 8c0 1.72.54 3.31 1.46 4.61L4.5 19.5l2.89-.96A7.95 7.95 0 0 0 12 20a8 8 0 0 0 8-8 8 8 0 0 0-8-8z"
                      fill="#5f4dd5"
                      opacity=""
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-136"
                />
              </button>
              <button
                className="MuiButtonBase-root-133 MuiButton-root-107 MuiButton-outlined-115 secondary-button sms-icon-button"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiButton-label-108"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-98 icon"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M0 0h24v24H0V0z"
                      fill="none"
                    />
                    <path
                      d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 14H6l-2 2V4h16v12z"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-136"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-4-33"
        >
          <div
            className="offer-selected"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="offer-best-match-selected"
            >
              Best Match
            </div>
            <div
              className="offer-detail-content"
            >
              <div
                className="offer-term"
              >
                36
                 months from
              </div>
              <div
                className="offer-price"
              >
                <div
                  className="offer-amount"
                >
                  Â£2,000
                </div>
                <div
                  className="offer-pm"
                >
                  /pm
                </div>
              </div>
            </div>
          </div>
          <div
            className="offer-unavailable"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="offer-detail-content"
            >
              <div
                className="offer-term"
              >
                48
                 months from
              </div>
              <div
                className="offer-price"
              >
                <div
                  className="offer-amount"
                >
                  Unavailable
                </div>
              </div>
            </div>
          </div>
          <div
            className="offer-detail"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="offer-detail-content"
            >
              <div
                className="offer-term"
              >
                60
                 months from
              </div>
              <div
                className="offer-price"
              >
                <div
                  className="offer-amount"
                >
                  Â£8,000
                </div>
                <div
                  className="offer-pm"
                >
                  /pm
                </div>
              </div>
            </div>
          </div>
          <div
            className="dealer-fee"
          >
            Excl. Â£200 dealer fee
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
